src/classes/speciessite.cpp

```diff
@@ -439,73 +372,103 @@ std::vector<std::shared_ptr<Site>> SpeciesSite::createFromParent() const
                 // If it's unique, remember it and proceed.
                 matchedIndices.push_back(std::move(matchedAtomIndices));
 
-                // Identifiers which label origin, x and y axis atoms.
-                auto identifiers = fragment_.matchedPath(&i).identifiers();
+                auto identifiers = matchedGroup.identifiers();
 
-                // Compute the origin.
-                auto originAtoms = identifiers["origin"];
+                // Determine origin atoms
+                sitesOriginAtoms_.push_back({identifiers["origin"].begin(), identifiers["origin"].end()});
 
-                if (originMassWeighted_)
-                {
-                    double massNorm = 0.0;
-                    for (const auto &atom : originAtoms)
-                    {
-                        mass = AtomicMass::mass(atom->Z());
-                        origin += atom->r() * mass;
-                        massNorm += mass;
-                    }
-                    origin /= massNorm;
-                }
-                else
-                {
-                    for (const auto &atom : originAtoms)
-                    {
-                        origin += atom->r();
-                    }
-                    origin /= originAtoms.size();
-                }
-
-                // Fragment site definition has orientation.
                 if (hasAxes())
                 {
+                    // Determine x axis atoms.
+                    sitesXAxisAtoms_.push_back({identifiers["x"].begin(), identifiers["x"].end()});
+                    // Determine y axis atoms.
+                    sitesYAxisAtoms_.push_back({identifiers["y"].begin(), identifiers["y"].end()});
+                }
+            }
+        }
+        return true;
+    }
+}
 
-                    auto xAxisAtoms = identifiers["x"];
-                    auto yAxisAtoms = identifiers["y"];
+// Return number of unique sites
+const int SpeciesSite::nSites() const { return type_ == SiteType::Static ? 1 : sitesOriginAtoms_.size(); }
 
-                    Vec3<double> v;
+const std::vector<std::vector<const SpeciesAtom *>> &SpeciesSite::sitesOriginAtoms() const { return sitesOriginAtoms_; }
```

Comment:
Need commented function description here and the others
```suggestion
// Return atoms representing unique site origins
const std::vector<std::vector<const SpeciesAtom *>> &SpeciesSite::sitesOriginAtoms() const { return sitesOriginAtoms_; }
```